### PR TITLE
Issue 2497 final

### LINKED ISSUE #2497: Panzer: Kokkos thread blocked gather/scatter
Right now the gather and scatters are only threaded for the standard DOF, and not the blocked one, EMPIRE uses the blocked one and this causes a big slowdown for real problems.  

@trilinos/panzer 

@jmgate @rppawlo is this one of you?

## Expectations

Ideally the gather/scatter code would get cleaned up, there is 20k lines of it and it seems hard to maintain. but at least thread the blocked G/S code.

## Current Behavior

Saw a massive slowdown in my code when I went to blocked assy, 

## Definition of Done

 - [ ] thread the blocked codes
## Steps to Reproduce

1. Open code in edittor Panzer_ScatterResidual_BlockedTpetra.hpp Panzer_GatherSolution_BlockedTpetra_impl.hpp
1. Shake fist angrily at computer.
est log file:  
* When was the last time everything worked (date/time; SHA1s; etc.)?
* What did you do that made the bug rear its ugly head?
* Have you tried turning it off and on again?
-->

### PR DESCRIPTION
**Pull request by rppawlo**
Kokkosify the blocked tpetra gather/scatter in #2497 
Tested on CUDA/P100, KNL and Broadwell.

Closes #2497 
@trilinos/panzer

### CODE REVIEW

File: packages/panzer/disc-fe/src/evaluators/Panzer_GatherSolution_BlockedTpetra.hpp [hunk 1]
```diff
@@ -147,11 +147,19 @@ class GatherSolution_BlockedTpetra<panzer::Traits::Residual,TRAITS,S,LO,GO,NodeT
   typedef Tpetra::Import<LO,GO,NodeT> ImportType;
   typedef Tpetra::Export<LO,GO,NodeT> ExportType;
 
-  // maps the local (field,element,basis) triplet to a global ID
-  // for scattering
-  Teuchos::RCP<const BlockedDOFManager<LO,GO> > gidIndexer_;
+  //! Maps the local (field,element,basis) triplet to a global ID for
+  // scattering
```
Comment (jmgate) [inline packages/panzer/disc-fe/src/evaluators/Panzer_GatherSolution_BlockedTpetra.hpp]: Is a ! needed on this second line?

Comment (rppawlo) [reply packages/panzer/disc-fe/src/evaluators/Panzer_GatherSolution_BlockedTpetra.hpp]: that adds the comment to doxygen

Comment (rppawlo) [reply packages/panzer/disc-fe/src/evaluators/Panzer_GatherSolution_BlockedTpetra.hpp]: sorry - not sure. Let me test

Comment (rppawlo) [reply packages/panzer/disc-fe/src/evaluators/Panzer_GatherSolution_BlockedTpetra.hpp]: yeah it needs it. will update after Eric reviews...

**Pull request merged**